src/pages/authpages/SignupPage.tsx

```diff
@@ -1,78 +1,309 @@
+            value={email}
+            onChange={handleInputChange}
           />
+          <div className="text-red-500">{emailError}</div>
           <Input
             type="text"
-            name="uname"
+            name="username"
             title="Enter your username"
+            borderColor={usernameError ? 'red' : '#E0E0E0'}
             placeholder="Username"
+            value={username}
+            onChange={handleInputChange}
           />
+          <div className="text-red-500">{usernameError}</div>
           <Input
             type="text"
-            name="fname"
+            name="firstname"
             title="Enter your firstname"
+            borderColor={firstnameError ? 'red' : '#E0E0E0'}
             placeholder="Firstname"
+            value={firstname}
+            onChange={handleInputChange}
           />
+          <div className="text-red-500">{firstnameError}</div>
           <Input
             type="text"
-            name="lname"
+            name="lastname"
             title="Enter your lastname"
+            borderColor={lastnameError ? 'red' : '#E0E0E0'}
             placeholder="Lastname"
+            value={lastname}
+            onChange={handleInputChange}
           />
+          <div className="text-red-500">{lastnameError}</div>
           <Input
             type="password"
             name="password"
             title="Enter your password"
+            borderColor={passwordError ? 'red' : '#E0E0E0'}
             placeholder="Password"
+            value={password}
+            onChange={handleInputChange}
           />
+          <div className="text-red-500">{passwordError}</div>
           <Button type="submit" title="Click to submit">
             Sign up with email
           </Button>
-          <Button type="button" title="Click to go back">
-            Go Back
-          </Button>
         </form>
-        <div className="flex flex-col gap-[18px] w-[60%] text-[#828282] items-center justify-center max-sm:gap-[8px]">
+        <div className="flex flex-col gap-[18px] w-[60%] text-[#828282] items-center justify-center max-sm:gap-[10px] max-md:gap-[14px] max-lg:gap-[16px]">
           <div className="flex items-center space-x-4">
             <div className="flex-grow border-b border-[#E6E6E6]"></div>
             <p className="text-center">or continue with</p>
             <div className="flex-grow border-b border-[#E6E6E6]"></div>
           </div>
-
-          <GoogleBtn />
+          <div className="flex items-center justify-center w-full">
+            <div
+              title="Click to continue with google"
+              className="bg-[#EEEEEE] w-full text-black rounded-[8px] py-2 px-4 box-border relative hover:bg-[#eceaea]"
+            >
+              <img
+                src={image2}
+                alt="google icon"
+                className="absolute top-[10px]"
+              />
+              <p>Google</p>
+            </div>
+          </div>
```

Comment:
<GoogleBtn />

Reply:
it's done you can review back @shebz2023